include/zephyr/senss/senss.h

```diff
@@ -0,0 +1,486 @@
+ *
+ * @param *handle The opened instance handle, if failed will be set to NULL.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_open_sensor(
+		int type, int sensor_index,
+		senss_sensor_handle_t *handle);
+
+/**
+ * @brief Close sensor instance.
+ *
+ * @param handle The sensor instance handle need to close.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_close_sensor(
+		senss_sensor_handle_t handle);
+
+
+/**
+ * @brief Set custom configuration from a sensor instance.
+ *
+ * Some sensor need support customized configuration, calibration data etc.
+ * The configuration data format is customized by usage.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param key The custom configuration item key.
+ *
+ * @param value The value buffer to be set.
+ *
+ * @param size The size of the value buffer in bytes.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_set_custom_config(
+		senss_sensor_handle_t handle,
+		enum senss_sensor_custom_config_key key,
+		void *value, int size);
+
+/**
+ * @brief Get custom configuration from a sensor instance.
+ *
+ * Some sensor need support customized configuration, calibration data etc.
+ * The configuration data format is customized by usage.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param key The custom configuration item key.
+ *
+ * @param value The value buffer to receive configuration value.
+ *
+ * @param size The size of the value buffer in bytes.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_get_custom_config(
+		senss_sensor_handle_t handle,
+		enum senss_sensor_custom_config_key key,
+		void *value, int size);
+
+/**
+ * @brief Get current report interval of given client handle.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param value The data buffer to receive value (micro seconds unit).
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_get_interval(
+		senss_sensor_handle_t handle,
+		uint32_t *value);
+
+/**
+ * @brief Set current report interval of given client handle.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param value The value to be set (micro seconds unit).
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+int senss_set_interval(
+		senss_sensor_handle_t handle,
+		uint32_t value);
+
+/**
+ * @brief Set current data change sensitivity of given client handle.
+ *
+ * Since each sensor type may have multiple data fields for it's value, this
+ * API support set separated sensitivity for each data field, or global
+ * sensitivity for all data fields.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param index The value fields index to be set, -1 for all fields (global).
+ *
+ * @param value The value to be set.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+int senss_set_sensitivity(
```

Comment:
`sensitivity` is a threshold value to control if the sample should be reported base on how much the sensor data changed, the rough logic is:  
```c
   if ((current_sample_data -  last_sample_data) >= sensitivity) {
       report data
   } else {
       not report data
   }
```
It's a standard `HID` concept, it's unit is per sensor data type, and different data fields of a sensor can have separated sensitivity values. 

It's a very useful config variable to control sensors, such as if set to 0, then meanings streaming every sample with fixed sample rate even no data changes, with a none zero number,  for example, 10 for a pedometer sensor, it's meanings every 10 steps will report a sample, 10, 20, 30, 40, ... etc.

Reply:
I see, thanks for that information, can this live in the application layer instead though? I 100% agree that there are use cases here even beyond HID, but this seems like something like something that's over complicating the subsystem. The reason I say this is that this is going to be needed to run per client of the subsystem as different clients might want to handle different sensitivities. It might be easier for clients to just implement this in their data callbacks.

Reply:
We can't live this in the Application layer,  since a virtual sensor also can set `sensitivity` to its reporters, but Application is totally dont' aware about a sensor is virtual or not and its reporters.

`senstivity` is very useful for a sensor subsystem, and it's not complicated, and this is a core value of the subsystem, with it, Applications is very easy to use, and can support many use cases.